GMX-Bots/Detect-many-closing/src/agent.ts

```diff
@@ -0,0 +1,57 @@
+import { NetworkData, networkData } from "./network";
+import { ethers, Finding, getEthersProvider, HandleTransaction, Log, TransactionEvent } from "forta-agent";
+import { NetworkManager } from "forta-agent-tools";
+
+import utils from "./utils";
+
+const networkManager = new NetworkManager(networkData);
+
+export const provideInitialize = () => {
+  return async () => {
+    const provider = getEthersProvider();
+    await networkManager.init(provider);
+  };
+};
+
+export const provideHandleTransaction =
+  (networkManager: NetworkManager<NetworkData>, provider: ethers.providers.JsonRpcProvider): HandleTransaction =>
+  async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
```

Comment:
Why this is `transactionHandler` instead of a `BlockHandler`? Your bot will be generating the same findings for every transaction in the block. 

Also, I know it's not very common to happen but let's add a check to make sure `fromBlock` is never negative, by setting it to `0` in that case. 

Another thought i have is that using `getLogs` with every new block doesn't seem to be the most optimized solution, we have a lot of block checked several times. Maybe executing this every `n` block instead? where `n` can be equal to `blockNumber` or set by the user. I would like to know your thoughts/ potential optimizations.